### PR TITLE
Rework che-theia tasks and commands

### PR DESCRIPTION
**Pull request by sunix**
Give a try: http://che.openshift.io/factory?url=https://github.com/eclipse/che-docs/tree/rework-che-commands

### CODE REVIEW

File: src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc [hunk 1]
```diff
@@ -18,78 +18,15 @@ summary:
 
 :context: defining-custom-commands-for-che-theia
 
-The Che-Theia IDE allows users to define custom commands in a devfile that are then available when working in a workspace.
+When starting a new project, new developers often loose time in finding the good procedure to build, run or debug a project. The result is documentation that are hard to follow and never up-to-date.
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc]: Hi @sunix, thanks for these updates. I believe these first two substitutions are using language that is better suited for a blogpost, fox example. The original wording was more in line with what is expected of docs.

File: src/main/pages/che-7/end-user-guide/con_che-theia-task-types.adoc [hunk 2]
```diff
@@ -3,36 +3,69 @@
...
+* The `command` field contains the command line for execution.
+* The `workdir` is the working directory in which the command is executed.
+* The `component` field refers to the container where the command will be executed. The field contains the component `alias` where the container is defined.
 
-* `theia:watch`, which runs the `theia` project in `watch` mode.
-* `theia:build`, which builds the `theia` project in the `che-dev` component.
 
+VS Code launch configurations:: `Attach remote debugger`
++
+VS Code launch configurations are most of the time used to defined debug configuration. These configurations could then be triggered when the developer hits `F5` or `Start Debugging` from the `Debug` menu. The configurations will give information to the debugger such as the port to connect for debugging or the type of application to debug (node, java, etc.).
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/con_che-theia-task-types.adoc]:
```suggestion
VS Code launch configurations are usually used to define debugging configuration. To trigger these configurations, press btn:[F5] or choose *Start Debugging* from the *Debug* menu. The configurations provide information to the debugger, such as the port to connect to for debugging or the type of the application to debug (Node.js, Java, etc.).
```

**Review by rkratky — CHANGES_REQUESTED** [review]
Thanks, @sunix. A couple of suggestions.

File: src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc (hunk 1 above)
Comment (sunix) [reply src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc]: Maybe, the goal is to give more context so users would understand when and why to use custom commands.

Comment (sunix) [reply src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc]: Previous wording was not helping in that purpose.

Comment (sunix) [reply src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc]: @rkratky is this fine for you?

Comment (amisevsk) [reply src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc]: To me it reads more like a marketing pitch than documentation.

Comment (sunix) [reply src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc]: if a marketing pitch helps the user to better understand why and when (the context) to use custom commands ... it is very fine to have that going to the doc :)

Comment (amisevsk) [reply src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc]: I disagree, when I look for documentation on a program I'm using, if I see ad copy I'm likely to never look at that documentation again. The documentation should answer the question "How do I add custom commands to my workspace?" -- if I'm asking the question I don't need some sentences justifying why it's a good idea.

Comment (sunix) [reply src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc]: This doc should answer the questions:
1. what is a che commands. including in which context a user should use it.
2. how do I add commands in my devfiles/workspaces

If it feels like marketing pitch for you, feel free to rephrase it. But it should include 1.

Comment (rkratky) [reply src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc]: @sunix I still disagree. We need to take into account that the majority of users who read documentation do not read all of it. They do not open it on page one and read it like a book until the end. They want to find a quick solution to their problem, or want to follow simple instructions on how to perform specific tasks.

The text that you're suggesting assumes that there would be readers who would want to read the documentation because of their interest in the application. You're describing the reasoning behind Che philosophy and offering insight into why certain features might be of interest to someone. That's important, but it's not the function of documentation. This would be for a product website, blogpost, presentation, or a magazine article.

Let's keep the documentation focused on specific tasks.

File: src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc [hunk 3]
```diff
@@ -18,78 +18,15 @@ summary:
...
 
-The following is an example of the `commands` section of a devfile.
+On boarding new developers is one of the main Che challenges. To help the new developers, lead developers can define devfiles to create on-demand workspaces for specific projects.
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc]: Again, I suggest leaving the orig. wording. If you believe we should better describe the purpose of devfiles, it should be included in the *What is a devfile* section at https://www.eclipse.org/che/docs/che-7/configuring-a-workspace-using-a-devfile/ (which, incidentally, could use a better title).

```suggestion
The following is an example of the `commands` section of a devfile.
```

File: src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc [hunk 4]
```diff
@@ -18,78 +18,15 @@ summary:
...
-              {
-               "type": "node",
-               "request": "attach",
-               "name": "Attach by Process ID",
-               "processId": "${command:PickProcess}"
-              }
-             ]
-            }
-----
+In the devfile, lead developers can provide pre-defined commands to help the team to execute common build, run or debug tasks for the project.
```
Comment (rkratky) [inline src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc]:
```suggestion

Pre-defined commands can help to execute common build, run, or debugging tasks for the project.
```

File: src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc (hunk 1 above)
Comment (sunix) [reply src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc]: @rkratky
> there would be readers who would want to read the documentation because of their interest in the application. 

Of course we have people reading the doc and read the doc because of their interest in the app!
They come to the documentation because they don't know the project and how to use it. With the current state of the doc, these users are just giving up because the docs is not helping them enough to understand and make good devfiles for their dev team.

Here we are describing how to use a feature but if we don't explain what is this feature about or for, it is useless.
This is not a troubleshooting section, but a user guide. These users don't know Che and what is the difference between VSCode and Che. To me this is very important.
OK this part could be somewhere else but it is important to link the custom commands to the Che "philosophy" and why custom commands in devfile would help to developers in the daily coding life.

BTW, this is just 3 lines as an intro/context, people could skip it and go to the right section if they want.

File: src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc (hunk 3 above)
Comment (amisevsk) [inline src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc]: The sentence

> On boarding new developers is one of the main Che challenges.

makes it sound like the documentation is saying "it's hard to start using Che"

Comment (sunix) [reply src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc]: s/challenges/purposes/ ?

File: src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc (hunk 1 above)
Comment (sunix) [reply src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc]: > We need to take into account that the majority of users who read documentation do not read all of it.

That's why recalling the context is important in this single page.

File: src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc (hunk 3 above)
Comment (amisevsk) [reply src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc]: Hm, I don't know that "purposes" works either, since then it's implying that Che is an onboarding tool.

Comment (sunix) [reply src/main/pages/che-7/end-user-guide/assembly_defining-custom-commands-for-che-theia.adoc]: Che is a good IDE for onboarding new developers in a project. Predefined commands is helping.

**Comment by sunix** [comment]
@rkratky @amisevsk
OK let's rework this. I am open to suggestions. 

But definitely, we can not leave orig wording, we need to give context and information to the person reading it:

 - who is this doc/page for:
     1. Team leads/Lead developers who will define the devfile the his team.
     2. Developers who would maintain or improve the devfile.
 - quick context about the devfile: What it is in 1 sentence  ... and links more generic section about the devfile (what it is, purpose, benefit for the team. How to create one, how to instance a workspace from it. How to share it with the team).
 - What is this page content: In this page, you will learn how to pre-define custom commands to a devfile. Custom commands in devfile will guide all the team members to execute common build, run or debug tasks for the project.

Why this is important: a user may not want to navigate through the whole doc to understand what it is about.
Before starting to read a page, a user is usually unsure that this page is what he is looking for. So having the context defined at the beginning should guide him so he knows he can continue reading or not.

**Review by rkratky — APPROVED** [review]
@sunix, thanks for all the input. I added some wording that seems like a reasonable from all POVs. I kept all your structural changes.

Merging this now to have an empty (content) PR queue for Antora migration. If you believe this should be revisited, please, open a new PR afterwards.
